client/src/main/java/io/mantisrx/server/master/client/MantisMasterClientApi.java

```diff
@@ -660,22 +677,37 @@ private Integer getSinkStageNumFromJsonResponse(String jobId, String response) t
                         .connect()
                         .flatMap((ObservableConnection<TextWebSocketFrame, TextWebSocketFrame> connection) -> connection.getInput()
                                 .map((TextWebSocketFrame webSocketFrame) -> webSocketFrame.text())))
-                .onErrorResumeNext(Observable.empty());
+                .onErrorResumeNext(Observable.empty())
+                .doOnCompleted(() -> {
+                  logger.info("Purging {} from job status cache", id);
+                  jobSchedulingInfoCache.remove(id);
+                })
+                .doOnUnsubscribe(() -> {
+                  logger.info("Purging {} from job status cache", id);
+                  jobSchedulingInfoCache.remove(id);
+                })
+                .replay(25)
```

Comment:
missing a `doOnError` to clear the cache? Wonder if that is causing the log line to not get printed in your testing.

Reply:
Interestingly the cache is evicted immediately with the new doOnError, but still no error message logged.